pkg/env/sensor.go

```diff
@@ -70,6 +70,13 @@ var (
 	// Multiple items can hold a pointer to the same object (e.g. same Deployment) so these numbers are pessimistic because we assume all items hold different objects.
 	DetectorNetworkFlowBufferSize = RegisterIntegerSetting("ROX_SENSOR_DETECTOR_NETWORK_FLOW_BUFFER_SIZE", 20000)
 
+	// DetectorDeploymentBufferSize indicates how many deployments will be kept in Sensor while offline in the detector.
+	// 1 Item in the buffer = ~1000 bytes
+	// 20000 * 1000 = 20 MB
+	// Notice: the actual size of each item is ~40 bytes since it holds pointers to the actual objects.
+	// Multiple items can hold a pointer to the same object (e.g. same Deployment) so these numbers are pessimistic because we assume all items hold different objects.
+	DetectorDeploymentBufferSize = RegisterIntegerSetting("ROX_SENSOR_DETECTOR_DEPLOYMENT_BUFFER_SIZE", 20000)
```

Comment:
We discussed this in the meeting and I agree that it is difficult to find a proper number here.
My 2 cents: I would rather err on the side of making this buffer too big than too small.

Making it too big:
- (+) current state is a buffer of infinite size
- (+) can cause OOMs and make the underlying problems visible
- (-) can cause OOMs

Making it too small:
- (+) makes sensor consume less memory (the same way as truncating all tables in db can make a system "faster"),
- (-) may hide potential problems (when messages are dropped)

Reply:
That makes sense. I do agree it's better to make them too big than too small. I think I'm going to do some testing with kube-burner and depending on the results I might bump the sizes.